RocketChat/Rocket.Chat
# [FIX] Fix error that occurs on changing archive status of room

## Checklist


## Proposed changes


This PR fixes an issue that happens when you try to edit the info of a room, and save changes after changing the value of "Archived". The archive functionality is handled separately from other room settings. The archived key is not used in the saveRoomSettings method but was still being sent over. Hence, the request was being considered invalid. I deleted the "archived" key from the data being sent in the request, making the request valid again.



## Issue(s)

Closes #20079 

## Steps to test or reproduce


- Edit Info of Room as a user.
- Change the value of "Archived"
- Click "Save Changes".
- "error-invalid-settings" error is shown

## Further comments

I considered not editing the data object, itself and instead of editing a copy of the data object, but it is itself being copied inside the function itself and does not affect anything else.
Another option that could be considered instead would be to manually unpack all the different values that can be present as in "client/views/admin/rooms/EditRoom.js", lines 128 to 140, but that might bring along unnecessary complexity.
